File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let currMin = prices[0];
    let currP = maxP = 0;

    for(let i=0;i<prices.length;i++) {

        if(prices[i]<currMin) currMin = prices[i];

        currP = prices[i]-currMin;

        // maxP = Math.max(currP,maxP);
        if(currP > maxP) maxP = currP;

    }
    return maxP;

    
};

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {

        int currMin = prices[0];
        int maxP = 0,currP = 0;

        for(int i=0;i<prices.length;i++){

            if(prices[i] <currMin) currMin = prices[i];
            currP = prices[i] - currMin;

            if(currP>maxP) maxP = currP;
            
        }

        return maxP;
        
    }
}